auth_tkt/ticket.py

```diff
@@ -29,19 +29,27 @@ def validate(ticket, secret, ip='0.0.0.0', timeout=7200, encoding='utf-8'):
 
     if '!' not in ticket:
         try:
-            raw = base64decode(ticket, encoding)
+            raw = base64decode(ticket, encoding=encoding)
             base64 = True
         except binascii.Error:
             return False
 
     if '!' not in raw:
         return False
 
-    raw = raw[32:]
+    if algo == 'SHA512':
+        raw = raw[128:]
+    elif algo == 'SHA256':
+        raw = raw[64:]
+    else:
```

Comment:
We could utilize `hashlib.algorithms` to check for supported hashing algorithms, then base the slice length off the value of `hashlib.new(algo).digest_size`?